src/dorsal.js

```diff
@@ -352,10 +356,15 @@ DorsalCore.prototype.wire = function(el, pluginName) {
         action;
 
     deferred.promise().progress(function() {
-        pluginsCalled++;
-        if (pluginsCalled === pluginCount) {
-            deferred.resolve();
+        var i = 0,
+            length = deferred.conditions.length;
+
+        for (; i < length; i++) {
+            if (deferred.conditions[i].state() !== 'resolved') {
+                return;
+            }
         }
+        deferred.resolve();
```

Comment:
We shouldn't need most of this actually. A promise could be the return value of `when`. We can probably just delete this entire progress section.